### PR TITLE
Pin multibuild scripts to get manylinux1 wheels back

### PR DESCRIPTION
**Pull request by jtattermusch**
Hotfix for https://github.com/protocolbuffers/protobuf/issues/9180.

See https://github.com/protocolbuffers/protobuf/issues/9180#issuecomment-966946662 for more context.

### CODE REVIEW

File: kokoro/release/python/linux/build_artifacts.sh [hunk 1]
```diff
@@ -25,7 +25,13 @@ rm -rf multibuild/
 mkdir artifacts
 export ARTIFACT_DIR=$(pwd)/artifacts
 
+# Pin multibuild script to a version just before the default
+# manylinux image has switched from manylinux1 to manylinux2014.
+# Also, pinning version avoid potentially unwanted future changes from
+# silently creeping in.
+# See https://github.com/protocolbuffers/protobuf/issues/9180
 git clone https://github.com/matthew-brett/multibuild.git
+(cd multibuild; git checkout 13a01725b0f0aa551ab34aa2311cdc1c77be4337)
```
Comment (jtattermusch) [inline kokoro/release/python/linux/build_artifacts.sh:34]: For future reference, the manylinux version can be set by setting 
`MB_ML_VERSION=1` when invoking the `build_wheel`.  But in our case it should only be done for x86_64 artifacts (aarch64 is only supported starting from manylinux2014)
https://github.com/matthew-brett/multibuild/blob/b89bb903e94308be79abefa4f436bf123ebb1313/travis_linux_steps.sh#L71

**Comment by haberman** [comment]
Thanks Jan!

It looks like the Linux Python Release job failed, perhaps because this version of multibuild doesn't support Python 3.10?

```
ls: /opt/python/cp310-cp310/bin: No such file or directory
```

**Comment by jtattermusch** [comment]
> Thanks Jan!
> 
> It looks like the Linux Python Release job failed, perhaps because this version of multibuild doesn't support Python 3.10?
> 
> ```
> ls: /opt/python/cp310-cp310/bin: No such file or directory
> ```

Ok. I updated the PR:

- I pinned multibuild at the last version that's currently available.
- now setting MB_ML_VER explicitly for linux artifacts.

**Comment by jtattermusch** [comment]
> Thanks Jan!
> 
> It looks like the Linux Python Release job failed, perhaps because this version of multibuild doesn't support Python 3.10?
> 
> ```
> ls: /opt/python/cp310-cp310/bin: No such file or directory
> ```

Ok, I know what the problem is. It's not about multibuild version.

The issue is that manylinux1  doesn't have python3.10 support (and I assume it will never have it because it's too old). 
The docker image `quay.io/pypa/manylinux1_x86_64:latest` only has python versions up to 3.9.

So for python3.10 we will need to opt for a newer version of manylinux image (which one? manylinux2010 or manylinux2014?)

**Comment by jtattermusch** [comment]
@acozzette  @haberman   which manylinux version should we use for python3.10? We need to decide on that otherwise this PR will be stuck indefinitely.

**Comment by acozzette** [comment]
I'm not very familiar with all this but I would vote for 2014 just because 2010 sounds pretty old at this point.

**Comment by haberman** [comment]
Can we try 2010? I think older is better here, since it increases our compatibility.

**Comment by acozzette** [comment]
2010 sounds good to me.

**Comment by jtattermusch** [comment]
I updated the python3.10 wheel build to use manylinux2010. Let's see what the tests say. If they pass, I think this PR is good to go.

**Comment by jtattermusch** [comment]
> Details

The wheels built by "Linux Python Release" job look good IMHO:
https://source.cloud.google.com/results/invocations/8f596554-2526-4504-9988-4123cff7b979/details

**Pull request merged**